File: dcp-solutions/2025-01-23/solution.rb
```ruby
class TreeNode
  attr_accessor :val, :left, :right
end

def min_path_sum(root)
  def dfs(node, path_sum)
    return path_sum if node.left.nil? && node.right.nil?
    left_sum = node.left ? dfs(node.left, path_sum + node.left.val) : nil
    right_sum = node.right ? dfs(node.right, path_sum + node.right.val) : nil
    [left_sum, right_sum].compact.sum || path_sum
  end
  return dfs(root, root.val)
end
```

File: dcp-solutions/2025-01-23/solution.php
```php
class TreeNode {
    public $val;
    public $left;
    public $right;

    function __construct($x) {
        $this->val = $x;
        $this->left = null;
        $this->right = null;
    }
}

function minPathSum($root) {
    if (!$root) return 0;

    if (!$root->left && !$root->right) {
        return $root->val;
    }

    $min_sum = PHP_INT_MAX;
    if ($root->left) {
        $sum = $this->minPathSum($root->left);
        $sum += $root->val;
        $min_sum = min($min_sum, $sum);
    }
    if ($root->right) {
        $sum = $this->minPathSum($root->right);
        $sum += $root->val;
        $min_sum = min($min_sum, $sum);
    }

    return $min_sum;
}
```